code/game/objects/breakable.dm

```diff
@@ -0,0 +1,298 @@
+		M.visible_message("<span class='warning'>\The [M] [M.attacktext] \the [src][generate_break_text(glanced,TRUE)]</span>","<span class='notice'>You hit \the [src][generate_break_text(glanced)]</span>")
+		try_break()
+	else
+		..()
+
+//Object ballistically colliding with something
+
+/obj/throw_impact(atom/impacted_atom, var/speed, mob/user)
+	..()
+	if(!(breakable_flags & BREAKABLE_AS_THROWN))
+		return
+	if(!(breakable_flags & BREAKABLE_MOB) && istype(impacted_atom, /mob)) //Don't break when it hits a mob if it's not flagged with BREAKABLE_MOB
+		return
+	if(isturf(loc)) //Don't take damage if it was caught mid-flight.
+		//Unless the object falls to the floor unobstructed, impacts happens twice, once when it hits the target, and once when it falls to the floor.
+		var/thisdmg = 10 * get_total_scaled_w_class(1) / (speed ? speed : 1) //impact damage scales with the weight class and speed of the object. since a smaller speed is faster, it's a divisor.
+		if(istype(impacted_atom, /turf/simulated/floor))
+			take_damage(thisdmg/2, skip_break = TRUE)
+		else
+			take_damage(thisdmg, skip_break = TRUE, mute = FALSE) //Be verbose about the object taking damage.
+		try_break(null, impacted_atom)
+
+//Object being hit by a projectile
+
+/obj/bullet_act(var/obj/item/projectile/proj)
+	..()
+	if(breakable_flags & BREAKABLE_WEAPON)
+		take_damage(proj.damage, skip_break = TRUE)
+	var/impact_power = max(0,round((proj.damage_type == BRUTE) * (proj.damage / 3 - (get_total_scaled_w_class(3))))) //The range of the impact-throw is increased by the damage of the projectile, and decreased by the total weight class of the object.
+	if(impact_power)
+		//Throw the object in the direction the projectile was traveling
+		var/propel_dir = get_dir(proj.starting, proj.target)
+		var/turf/T = get_edge_target_turf(loc, propel_dir)
+		throw_at(T, impact_power, proj.projectile_speed)
+	else if(breakable_flags & BREAKABLE_WEAPON)
+		try_break()
+
+/obj/proc/get_total_scaled_w_class(var/scalepower=3) //Returns a scaled sum of the weight class of the object itself and all of its contents, if any.
+	//scalepower raises the w_class of each object to that exponent before adding it to the total. This helps avoid things like a container full of tiny objects being heavier than it should.
+	var/total_w_class = (isnull(w_class) ? W_CLASS_MEDIUM : w_class) ** scalepower
+	if(!isnull(contents) && contents.len)
+		for(var/obj/item/thiscontent in contents)
+			total_w_class += (thiscontent.w_class ** scalepower)
+	return total_w_class
+
+//Biting the object
+
+/obj/bite_act(mob/living/carbon/human/biter)
+	if(breakable_flags & BREAKABLE_UNARMED && biter.can_bite(src))
+		var/thisdmg = BREAKARMOR_FLIMSY
+		if(biter.organ_has_mutation(LIMB_HEAD, M_BEAK)) //Beaks = stronger bites
+			thisdmg += 4
+
+		var/attacktype = "bite"
+		var/attacktype2 = "bites"
+		var/datum/butchering_product/teeth/T = locate(/datum/butchering_product/teeth) in biter.butchering_drops
+
+		if(T.amount == 0)
```

Comment:
Can T be null here?

Reply:
I don't think so, because apparently locate(_type_) works the same way as istype(), returning FALSE if they are different types.
Also I copied that from the already existing biting-a-mob code in human_attackhand.dm.

Reply:
I think locate(_type_) returns FALSE if there is nothing found. But I added a check to be safe.
I originally copied this part from human_attackhand.dm.